dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -631,6 +632,57 @@ public ListenableFuture<List<RouteFailure>> copy(List<Route> routes, UfsReadOpti
     return Futures.whenAllComplete(futures).call(() -> errors, GrpcExecutors.BLOCK_WRITER_EXECUTOR);
   }
 
+  @Override
+  public ListenableFuture<List<RouteFailure>> move(List<Route> routes, UfsReadOptions readOptions,
+                                                   WriteOptions writeOptions) {
+    List<ListenableFuture<Void>> futures = new ArrayList<>();
+    List<RouteFailure> errors = Collections.synchronizedList(new ArrayList<>());
+    for (Route route : routes) {
+      UnderFileSystem srcUfs = mUfsManager.getOrAdd(new AlluxioURI(route.getSrc()),
+              UnderFileSystemConfiguration.defaults(mConf));
+      String srcRoot = new AlluxioURI(route.getSrc()).getRootPath();
+      String dstRoot = new AlluxioURI(route.getDst()).getRootPath();
+      UnderFileSystem dstUfs = mUfsManager.getOrAdd(new AlluxioURI(route.getDst()),
+              UnderFileSystemConfiguration.defaults(mConf));
+
+      try (FileSystem srcFs = new UfsBaseFileSystem(mFsContext, new UfsFileSystemOptions(srcRoot),
+              new UfsManager.UfsClient(() -> srcUfs, new AlluxioURI(srcRoot)));
+           FileSystem dstFs = new UfsBaseFileSystem(mFsContext, new UfsFileSystemOptions(dstRoot),
+                   new UfsManager.UfsClient(() -> dstUfs, new AlluxioURI(dstRoot)))) {
+        ListenableFuture<Void> future = Futures.submit(() -> {
+          Boolean deleteFailure = false;
+          try {
+            if (readOptions.hasUser()) {
+              AuthenticatedClientUser.set(readOptions.getUser());
+            }
+            CopyHandler.copy(route, writeOptions, srcFs, dstFs);
+            try {
+              DeleteHandler.delete(new AlluxioURI(route.getSrc()), srcFs);
+            } catch (Exception e) {
+              deleteFailure = true;
+              throw e;
+            }
+          } catch (Exception t) {
+            AlluxioRuntimeException e = AlluxioRuntimeException.from(t);
+            RouteFailure.Builder builder =
+                RouteFailure.newBuilder().setRoute(route).setCode(e.getStatus().getCode().value());
+            if (e.getMessage() != null) {
+              builder.setMessage(e.getMessage());
+            }
+            if (deleteFailure) {
+              builder.setRetryable(false);
```

Comment:
shouldn't be if there's delete failure it is retryable, but if there's no delete failure but failed at copy it is not retryable?
And also if no deletefailure the retryable is originally false too.

Reply:
and i think u can also distinguish the exception into alluxioruntimeexception to determine if this deletion failure is retryable or not right?

Reply:
@lucyge2022  By default, the InternalRuntimeException is unretryable. So adding a flag here will be an easy way to distinguish whether the failure comes from delete or not. A deletion failure will not be retryable, because we already have retried delete in the DeleteHandler.

Reply:
understood